kodilla-patterns/src/main/java/com/kodilla/patterns/prototype/library/Library.java

```diff
@@ -0,0 +1,39 @@
+package com.kodilla.patterns.prototype.library;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public final class Library extends Prototype {
+
+    String name;
+    Set<Book> books = new HashSet<>();
+
+    public Library(final String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public Set<Book> getBooks() {
+        return books;
+    }
+
+    public Library shallowCopy() throws CloneNotSupportedException {
+        return (Library) super.clone();
+    }
+
+    public Library deepCopy() throws CloneNotSupportedException {
+        Library clonedLibrary = (Library)super.clone();
+        clonedLibrary.books = new HashSet<>();
+        for(Book theSet : books) {
+            clonedLibrary.books.add(theSet);
+        }
```

Comment:
Zamiast
        for(Book theSet : books) {
            clonedLibrary.books.add(theSet);
        }
Mozna zapisac ktorej
clonedLibrary.books.addAll(books);